### PR TITLE
update knex examples

### PR DESCRIPTION
**Pull request by caub**
just adding more variants

### CODE REVIEW

File: examples/Knex.md [hunk 1]
```diff
@@ -13,15 +13,22 @@ const DataLoader = require('dataloader');
...
+    // .then(rows => {
+    //   const m = new Map(rows.map(row => [row.id, row]));
+    //   return ids.map(id => m.get(id));
+    // })
 
   story: new DataLoader(ids => db.table('stories')
-    .whereIn('id', ids).select()
-    .then(rows => ids.map(id => rows.find(x => x.id === id)))),
+    .whereIn('id', ids)
+    .orderByRaw('array_position(:ids, id)', { ids })
```
Comment (koistya) [inline examples/Knex.md]: I assume this syntax is specific to Postgres, right?

Comment (koistya) [reply examples/Knex.md]: ..in the case with Postgres, you can also consider the `with ordinality` syntax demonstrated here https://stackoverflow.com/questions/866465/order-by-the-in-value-list/35456954#35456954

Comment (caub) [reply examples/Knex.md]: Thanks good point, I updated it

**Comment by koistya** [comment]
@caub do you have any benchmarks of which method works faster (assuming that normally there are no more than 100 keys being passed to a data loader).

One advantage of mapping rows locally is that you can use the same syntax for both 1-to-1 and 1-to-many mappings:

```js
  commentById = new DataLoader(keys =>
    db
      .table('comments')
      .whereIn('id', keys)
      .select()
      .then(mapTo(keys, x => x.id)),
  );

  commentsByStoryId = new DataLoader(keys =>
    db
      .table('comments')
      .whereIn('story_id', keys)
      .select()
      .then(mapToMany(keys, x => x.story_id)),
  );
```

https://github.com/kriasoft/nodejs-api-starter/blob/3612dc4/src/Context.js#L94-L100
https://github.com/kriasoft/nodejs-api-starter/blob/3612dc4/src/Context.js#L102-L108

**Comment by caub** [comment]
@koistya good idea, because it's not what I expected https://gist.github.com/caub/3ce61e9ad8a18d74e39f2ab419a6d388

**Comment by leebyron** [comment]
Thanks for this thoughtful investigation, it's especially insightful that local mapping is faster than using the ordinality attribute in the query itself.

I'm going to close this PR though, since the goal of these examples is to be explanatory and not necessarily to find the best micro-performance. I think it's probably not the place for working out the fastest implementation of id mapping
